### ISSUE TITLE
Autocompletion for "ssh -F config someth<TAB>" is broken

### ISSUE THREAD

**Issue by jruere** [fish-shell/fish-shell#6905]
fish, version 3.1.0
3.1.0
Linux serpico 5.6.4-arch1-1 #1 SMP PREEMPT Mon, 13 Apr 2020 12:21:19 +0000 x86_64 GNU/Linux
xterm-256color

It completes some command instead of a host.

Using `sh -c 'env HOME=$(mktemp -d) fish'` has the same behaviour.

**Comment by faho** [comment]
Yeah, the ssh completion needs to know about the actual option arguments.

In general this is a hard problem - in fact it's impossible without knowing the actual options.

In particular it means that the ssh completions need to be altered to use `argparse`.

**Comment by jruere** [comment]
This used to work in 2.something. Do I remember incorrectly?

**Comment by faho** [comment]
>Do I remember incorrectly?

Yes.

**Comment by RaafatTurki** [comment]
hi, any update?

**Comment by stratacast** [comment]
Similar issue with `ssh -O`. For example, `ssh -O exit example.com`. fish does not autocomplete the list of hosts

**Comment by krobelus** [comment]
> Similar issue with `ssh -O`. For example, `ssh -O exit example.com`. fish does not autocomplete the list of hosts

right, we should make it use argparse, like in share/completions/git.fish
